### PR TITLE
Dashboard: Overflow and z-index adjustments

### PR DESCRIPTION
**Pull request by leeoniya**
this extracts some needed adjustments from the tooltips PR: https://github.com/grafana/grafana/pull/70872

the main thing we need to verify is that it doesn't break existing stuff. primarily overflow in things that which **SplitPane** and/or **SplitPaneWrapper**, such as Explore and Panel Edit.

specifically, this undoes the `overflow: hidden` style of the external `SplitPane` component. this allows a tooltip portalled into a panel within the SplitPane to render over the panel edit sidebar or into the adjacent (right) split pane in Explore.

previous/related changes:

https://github.com/grafana/grafana/pull/75499
https://github.com/grafana/grafana/pull/75382
https://github.com/grafana/grafana/pull/75328

### CODE REVIEW

**Comment by leeoniya** [comment]
still needs work...

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 1]
```diff
@@ -358,6 +358,15 @@ const getStyles = (theme: GrafanaTheme2) => {
...
       flexDirection: 'column',
 
+      '> *': {
+        zIndex: 0,
+      },
+
+      // matches .react-grid-item styles in _dashboard_grid.scss to ensure any contained tooltips occlude adjacent panels
+      '&:hover, &:active, &:focus': {
+        zIndex: 999,
+      },
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:368]: this handles the case then the panel is not in a grid, such as via PanelRenderer

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 2]
```diff
@@ -358,6 +358,15 @@ const getStyles = (theme: GrafanaTheme2) => {
       display: 'flex',
       flexDirection: 'column',
 
+      '> *': {
+        zIndex: 0,
+      },
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:363]: this prevents tooltips from going under things like RadioButtonGroup such as the ones in those in the Explore graph header, where the options have explicit z-index of 1 or 3. we also have explicit z-index on form Input elements which makes them render over the tooltip when it overlaps an adjacent panel.

File: public/app/core/components/SplitPaneWrapper/SplitPaneWrapper.tsx [hunk 3]
```diff
@@ -141,6 +142,9 @@ const getStyles = (theme: GrafanaTheme2, hasSplit: boolean) => {
   `;
 
   return {
+    splitPane: css`
+      overflow: visible !important;
```
Comment (leeoniya) [inline public/app/core/components/SplitPaneWrapper/SplitPaneWrapper.tsx:146]: in panel edit, this allows the tooltip to exit the splitpane with the viz and render on top of the options sidebar. it doesnt have the desired effect in Explore, because that has layers of `overflow` rules from the `react-custom-scrollbars`. this will be solved differently in another PR.

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx (hunk 1 above)
Comment (axelavargas) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:368]: 🤔 it seems this change conflicts with the ability to resize the panel using drag a drop. 

https://github.com/grafana/grafana/assets/239999/314d5432-7265-4adf-aad2-915fb6d7527d

Comment (leeoniya) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:368]: ah, good catch. i wonder what the z-index is of that drag handle :thinking:

Comment (leeoniya) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:368]: fixed in [b3dcbe8](https://github.com/grafana/grafana/pull/75597/commits/b3dcbe8ee05c82104df7028689567045b565bedd)

**Comment by Elfo404** [comment]
to me this works also in Explore (Chrome, FF - mac), is https://github.com/grafana/grafana/pull/75597#discussion_r1340520265 still relevant? 

https://github.com/grafana/grafana/assets/1170767/b258bf3a-da25-4add-a8de-616ebdceeb46

the only "weird" thing is the scrollbar appearing above the tooltip (which doesn't happen in main).

File: public/sass/components/_dashboard_grid.scss [hunk 4]
```diff
@@ -13,6 +13,8 @@
   &:hover {
     .react-resizable-handle {
       visibility: visible;
+      // above other panels, but below lowest pre-defined index in zIndex.ts
+      z-index: 999;
```
Comment (axelavargas) [inline public/sass/components/_dashboard_grid.scss:17]: Do you think we would benefit from defining this `z-index` value in `zIndex.ts`? It seems that the value is context-dependent, and I am concerned that if that file changes its values, this element will no longer be below the lowest pre-defined index.

**Review by axelavargas — APPROVED** [review]
Hey @leeoniya 👋🏾 , 

I double-checked the issue regarding drag and drop and did additional testing. Besides Gio's comment, I could not find anything else 🏅 

P.S I left a minor comment :)

File: public/sass/components/_dashboard_grid.scss (hunk 4 above)
Comment (leeoniya) [reply public/sass/components/_dashboard_grid.scss:17]: i wanted to put it there, but how do we import it into scss files? 🤔

Comment (axelavargas) [reply public/sass/components/_dashboard_grid.scss:17]: 🤔 that's a great question, it seems we already do that in other places https://github.com/grafana/grafana/blob/main/public/sass/components/_dashboard_settings.scss#L9, we are mapping them here, but tbh I am not very familiar with the whole themes https://github.com/grafana/grafana/blob/6614eb0a6e8cf2ccce55bec2d99bf4e33601cf46/packages/grafana-ui/src/themes/_variables.scss.tmpl.ts#L164

Comment (leeoniya) [reply public/sass/components/_dashboard_grid.scss:17]: ah, nice. i'll do that then 👍

**Comment by leeoniya** [comment]
> to me this works also in Explore (Chrome, FF - mac), is https://github.com/grafana/grafana/pull/75597#discussion_r1340520265 still relevant?
> the only "weird" thing is the scrollbar appearing above the tooltip (which doesn't happen in main).

@Elfo404 yes it's still relevant (though i cannot repro it in Linux/Chrome). unfortunately i dont think this can be fixed easily without touching `react-custom-scrollbars-2` styling, which is a big can of worms that i don't want to touch here (or at all).

the way we'll aim to solve this in a future PR (once we portal tooltips into panels) is to treat any `react-custom-scrollbars-2` container as the bounding box for tooltip positioning. so the tooltip would shift to the left side of the cursor before it overlaps the scrollbar. we will also play around with `position: fixed`, but that brings back the same scroll-related positioning headaches as the current approach of a global portalling.

btw, did you have to manually switch out the hardcoded tooltip mode to `TooltipDisplayMode.Multi`? i'm having trouble getting the multi-tooltip to show up in Explore as in your screen rec. my screenshot uses the new tooltip model which still ignores the single/multi setting :)

**Comment by Elfo404** [comment]
@leeoniya yep, there's no hidden setting :)
